### ISSUE TITLE
[Feature Request} Add pagination / filter to DNS Server Zone Addresses

### ISSUE THREAD

**Issue by mgutt** [webmin/webmin#2206]
I'm having a zone with more than 1000 addresses and the browser (especially Google Chrome) has huge problems displaying so much content. That's why I like to see a pagination and filter option, so for example only 100 addresses per page are displayed.

**Comment by iliajie** [comment]
Hello,

It's already available in the Authentic theme, e.g.:

<img width="926" alt="image" src="https://github.com/webmin/webmin/assets/4426533/d0ac2773-33c2-4445-8e6f-d2294a85e112">

**Comment by mgutt** [comment]
You are showing the pagination of the File Manager but I need a pagination for the Bind DNS Server > Zone > Adresses page

I'm already using the Authentic theme with the current Webmin 2.111, but I'm not sure if the theme is maybe an older version.

**Comment by iliajie** [comment]
The BIND module configuration has a few related options in the **Display options** subpage for **"Display domains as"** and **"Maximum number of zones to display"**. However, I don't recall it having a new pagination API.

If you could send me a config file with all those zones, I can reproduce the view and have a working case to try implementing this before the next Webmin release. You can email it to me privately by using my domain ross dot gdn with ilia as a user.

**Comment by jcameron** [comment]
There is an option on the module configuration page for the BIND module to not show records in zones by default, and instead just show a search box.

**Comment by iliajie** [comment]
> There is an option on the module configuration page for the BIND module to not show records in zones by default, and instead just show a search box.

Yeah, though some time ago I created a new pagination API that we should use instead (as we do in **Users and Groups** module). Relying on a search box without displaying any results isn't the best solution nowadays.

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run.

⚠️ If the issue is resolved, please take a moment to close it to help keep the project organized.
